### PR TITLE
LPS-97199

### PR DESCRIPTION
**Pull request by dgarciasarai**
/cc @darquesdev

### CODE REVIEW

File: modules/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/processor/AsahSegmentsExperimentsProcessor.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+		sb.append(_portal.getPathFriendlyURLPublic());
+
+		sb.append(group.getFriendlyURL());
+		sb.append(layout.getFriendlyURL());
+
+		return sb.toString();
+	}
+
+	private ExperimentStatus _toDTO(int status) {
+		if (status == SegmentsConstants.SEGMENTS_EXPERIMENT_STATUS_DRAFT) {
```
Comment (dgarciasarai) [inline modules/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/processor/AsahSegmentsExperimentsProcessor.java]: Review this

**Comment by epgarcia** [comment]
ci:test:sf

File: modules/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/processor/AsahSegmentsExperimentsProcessor.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+			experiment.setDXPSegmentName(
+				segmentsEntry.getName(LocaleUtil.getDefault()));
+		}
+
+		experiment.setExperimentStatus(_toDTO(segmentsExperiment.getStatus()));
+
+		Layout layout = _layoutLocalService.getLayout(
+			segmentsExperiment.getClassPK());
+
+		experiment.setPageURL(_getLayoutFullURL(layout));
```
Comment (dgarciasarai) [inline modules/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/processor/AsahSegmentsExperimentsProcessor.java]: We are using now full URL

**Comment by epgarcia** [comment]
ci:test:relevant

File: modules/apps/segments/segments-api/src/main/java/com/liferay/segments/constants/SegmentsConstants.java [hunk 3]
```diff
@@ -29,11 +29,15 @@
 
 	public static final long SEGMENTS_ENTRY_ID_DEFAULT = 0;
 
+	public static final String SEGMENTS_ENTRY_KEY_DEFAULT = "DEFAULT";
```
Comment (dgarciasarai) [inline modules/apps/segments/segments-api/src/main/java/com/liferay/segments/constants/SegmentsConstants.java]: New keys

File: modules/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/processor/AsahSegmentsExperimentsProcessor.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+		Experiment experiment = new Experiment();
+
+		experiment.setCreateDate(segmentsExperiment.getCreateDate());
+		experiment.setModifiedDate(segmentsExperiment.getModifiedDate());
+		experiment.setName(segmentsExperiment.getName());
+		experiment.setDataSourceId(_asahFaroBackendClient.getDataSourceId());
+		experiment.setDescription(segmentsExperiment.getDescription());
+
+		if (segmentsExperiment.getSegmentsExperienceId() ==
```
Comment (dgarciasarai) [inline modules/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/processor/AsahSegmentsExperimentsProcessor.java]: If we are creating an Experiment for the default Experience, we use the default Experience and default Segment information

File: modules/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/processor/AsahSegmentsExperimentsProcessor.java [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Sarai Díaz
+ * @author David Arques
+ */
+@Component(immediate = true, service = AsahSegmentsExperimentsProcessor.class)
+public class AsahSegmentsExperimentsProcessor {
```
Comment (darquesdev) [inline modules/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/processor/AsahSegmentsExperimentsProcessor.java]: Should it be singular?

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by darquesdev** [comment]
@epgarcia please, review this [PR](https://github.com/epgarcia/liferay-portal/pull/862) instead

**Pull request closed without merging**